include/RevThread.h

```diff
@@ -76,58 +46,100 @@ class RevThread {
     }
   }
 
-  // ThreadID operations
-  uint32_t GetThreadID() const { return ThreadID; }
-  void SetThreadID(const uint32_t NewThreadID) { ThreadID = NewThreadID; }
-  uint32_t GetParentThreadID() const { return ParentThreadID; }
-  void SetParentThreadID(const uint32_t NewParentThreadID) { ParentThreadID = NewParentThreadID; }
+  ///< RevThread: Get this thread's ID
+  uint32_t GetID() const { return ID; }
+
+  ///< RevThread: Set this thread's ID
+  void SetID(const uint32_t NewID) { ID = NewID; }
+
+  ///< RevThread: Get the parent of this thread's TID
+  uint32_t GetParentID() const { return ParentID; }
+
+  ///< RevThread: Set the parent of this thread's TID
+  void SetParentID(const uint32_t NewParentID) { ParentID = NewParentID; }
+
+  ///< RevThread: Get the TID of the thread that this thread is waiting to join
   uint32_t GetWaitingToJoinTID() const { return WaitingToJoinTID; }
+
+  ///< RevThread: Set the TID of the thread that this thread is waiting to join
   void SetWaitingToJoinTID(const uint32_t ThreadToWaitOn) { WaitingToJoinTID = ThreadToWaitOn; }
 
-  // Add new file descriptor
+  ///< RevThread: Add new file descriptor to this thread (ie. rev_open)
   void AddFD(int fd){ fildes.insert(fd); }
 
-  // Remove file descriptor from this thread (ie. rev_close)
+  ///< RevThread: Remove file descriptor from this thread (ie. rev_close)
   void RemoveFD(int fd){ fildes.erase(fd); }
 
-  // See if file descriptor exists/is owned by this thread
+  ///< See if file descriptor exists/is owned by this thread
   bool FindFD(int fd){ return fildes.count(fd); }
 
-  // Get the threads
+  // @DAVE: Potentially move these to RevHart and have them set it when it's assigned?
+  // Arguments could be made to keep it here or put it there
+  void SetLSQueue(std::shared_ptr<std::unordered_map<uint64_t, MemReq>> lsq){
+    VirtRegState->SetLSQueue(lsq);
+  }
+
+  // @DAVE: Potentially move these to RevHart and have them set it when it's assigned?
+  // Arguments could be made to keep it here or put it there
+  void SetMarkLoadComplete(std::function<void(const MemReq&)> func){
+    VirtRegState->SetMarkLoadComplete(func);
+  }
```

Comment:
A LSQ manages requests for multiple threads/harts but is tied to a single Proc, and as a Hart is also tied to a Proc it may make more sense to put it in RevHart, especially if threads can migrate between Procs - having LSQ in RevHart would remove the chance of a thread ever referencing a LSQ outside the Proc it is are running on.

Reply:
Moved... the LSQueue & MarkLoadComplete are now parts of the RevHart constructor. Everything passes but please review the new implementation in RevHart